### ISSUE TITLE
error with liblua5.2.so.0 on Fedora Linux

### ISSUE THREAD

**Issue by techtonik** [CleverRaven/Cataclysm-DDA#26745]
**Describe the bug**  
```
➜  cataclysmdda-0.C ./cataclysm                                                                                  
./cataclysm: error while loading shared libraries: liblua5.2.so.0: cannot open shared object file: No such file or directory
➜  cataclysmdda-0.C sudo dnf install lua                     
Last metadata expiration check: 0:00:00 ago on Sun 18 Nov 2018 04:33:04 AM +03.
Package lua-5.3.4-10.fc28.x86_64 is already installed, skipping.
```

**To Reproduce**  
1. Just run the game downloaded from http://dev.narc.ro/cataclysm/jenkins-latest/Linux_x64/Curses/cataclysmdda-0.C-8186.tar.gz

**Versions and configuration(please complete the following information):**
 - OS: Fedora 28
 - Graphics version Terminal

**Comment by ZhilkinSerg** [comment]
I guess you need to install Lua 5.2 (i.e. `sudo dnf install lua5.2`).

**Comment by techtonik** [comment]
@ZhilkinSerg nope, there are no such old packages - https://apps.fedoraproject.org/packages/s/lua5

**Comment by ZhilkinSerg** [comment]
Then you are out of luck using precompiled binary until we switch to using Lua 5.3 on Jenkins. @narc0tiq, can you upgrade Lua version for Linux builds? 

Also see how somebody fixed that - https://discourse.cataclysmdda.org/t/problem-with-missing-liblua5-2-so-0-on-latest-linux-x64-build-on-fedora-28/15774

**Comment by ZhilkinSerg** [comment]
Then you are out of luck using precompiled version.

**Comment by narc0tiq** [comment]
Sure, I've installed `lua5.3` and `liblua5.3-dev`, but I'm not sure of the Makefile changes (if any) required to use it. Actually, on a quick peek, it looks like it'll just get used automatically?

**Comment by ZhilkinSerg** [comment]
We don't define lua version explicitly in the makefile, so it should work fine.

**Comment by narc0tiq** [comment]
Hm? We do have some code that searches for Lua availability: https://github.com/CleverRaven/Cataclysm-DDA/blob/771c9e47267a75c57459b7137000bcfa11655afd/Makefile#L512-L519

Unless I'm misunderstanding what that does?

**Comment by techtonik** [comment]
@narc0tiq so when you install `lua5.3` what does `ldd cataclysm` shows on rebuilt binary?

**Comment by narc0tiq** [comment]
Yes, builds since [that earlier comment](https://github.com/CleverRaven/Cataclysm-DDA/issues/26745#issuecomment-440216265) have been running against lua5.3.

Static vs dynamic is up to the Makefile, I'm sure there's been some discussion on the subject before. My personal complaint would be that the output binary would become _much_ bigger, and it's already a pain to maintain the build history going back as far as it does.

**Comment by techtonik** [comment]
Still doesn't run. The library is called `/usr/lib64/liblua-5.3.so` on Fedora 28.

https://rpmfind.net/linux/RPM/fedora/28/x86_64/l/lua-libs-5.3.4-10.fc28.i686.html

**Comment by ZhilkinSerg** [comment]
> Hm? We do have some code that searches for Lua availability:

Yeah, it seems it searches for the most recent version.

**Comment by techtonik** [comment]
So what is the Lua overhead? Why it is not possible to compile it statically?

**Comment by ZhilkinSerg** [comment]
Closed with #28572.
